src/modelfree/policies/wrappers.py

```diff
@@ -36,3 +36,44 @@ def predict(self, observation, state=None, mask=None, deterministic=False):
         noise = self.noise_generator(noise_param, action_shape)
         noisy_actions = original_actions * (1 + noise)
         return noisy_actions, states
+
+
+class MultiPolicyWrapper(DummyModel):
+    def __init__(self, policies, num_envs):
+        # TODO how do we do this properly, since DummyModel requires a single policy and sess?
+        super().__init__(policies, policies[0].sess)
+        self.policies = policies
+        # Can't currently easily read n_env off policies because they differ in n_env vs n_envs
+        self.num_envs = num_envs
+        self.action_space_shape = self.policies[0].policy.action_space.shape
+        for p in self.policies:
+            assert p.policy.action_space.shape == self.action_space_shape, "All policies must " \
+                                                                           "have the same" \
+                                                                           " action space"
+
+        self.current_env_policies = np.random.choice(self.policies, size=self.num_envs)
+
+    def predict(self, observation, state=None, mask=None, deterministic=False):
+        self.reset_current_policies(mask)
+        policy_actions = np.empty(shape=(self.num_envs,) + self.action_space_shape)
```

Comment:
I think you can just change `self.num_envs` here to be `observation.shape[0]` and you'll be OK.

Having said that, we probably should patch gym compete to be consistent with stable baselines policies with the `n_envs` attribute.

Reply:
I think I'm inclined to keep `self.num_envs` here, since that makes it clearer what's going on, and I think we actually want to have `num_envs` around at initialization anyway so that we can construct `self.current_env_policies`,  which needs one entry per env, at that point. 

(We technically could wait to construct/assign self.current_env_policies until we get our first observation, but that feels somewhat messy to have to have `self.current_env_policies` be initially None and then only take a value in the middle of our first predict loop)